### PR TITLE
Add reference section to all statistics examples

### PR DESCRIPTION
**Pull request by ianhi**
## PR Summary
As the title says.

I didn't add every function used in the examples,  rather I erred on the side of caution against overlinking things such as `set_ylabel` and tried to include only the functions that the examples focus on.

## PR Checklist

- [NA] Has pytest style unit tests (and `pytest` passes).
- [X] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [NA] New features are documented, with examples if plot related.
- [NA] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [NA] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [NA] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [NA] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by ianhi** [comment]
Three other thoughts:
1. There are a **lot** of boxplot ex
2. `bxp` and `boxplot` never have their difference coherently explained as far as I can tell
3. Are PRs like this one desireable? Better linking in the docs seems like a positive, but I don't want to add uncesssary noise

**Comment by jklymak** [comment]
This is very welcome.  We just haven't gotten around to doing every example yet and have been counting on people just adding them as needed.

**Comment by timhoffm** [comment]
> I didn't add every function used in the examples, rather I erred on the side of caution against overlinking things such as set_ylabel and tried to include only the functions that the examples focus on.

:+1:

> 1. There are a **lot** of boxplot examples

There's quite some redundancy in some areas of the examples; I guess for historic reasons. More and more examples were added and it was not always checked if this adds something new or if there should have been some rearrangement. A bit consolidation would be good. But that's also a tedious process. One has to find out what the core message of each example is and if that's already explained in another example. If you see something that is duplicate or if some examples could be merged easily, you're welcome to open a PR.

> 3. Are PRs like this one desireable? Better linking in the docs seems like a positive, but I don't want to add uncesssary noise

Yes definitively. The references are value in particular because of the backlinks from the functions to the examples. But choosing the relevant references must be done manually.

File: examples/statistics/confidence_ellipse.py [hunk 1]
```diff
@@ -214,3 +214,15 @@ def get_correlated_dataset(n, dependency, mu, scale):
...
+#
+# ------------
+#
+# References
+# """"""""""
+#
+# The use of the following functions and methods is shown in this example:
+
+import matplotlib
+matplotlib.transforms.Affine2D
```
Comment (timhoffm) [inline examples/statistics/confidence_ellipse.py:228]: I'd add `matplotlib.patches.Ellipse`

**Review by timhoffm — COMMENTED** [review]
Generally, we add the corresponding pyplot functions as well, even if they are not used. See e.g. https://51381-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/statistics/barchart_demo.html#sphx-glr-gallery-statistics-barchart-demo-py

This is quite a crutch, but the only way, that the pyplot functions get to show the examples as well.

**Comment by ianhi** [comment]
> Generally, we add the corresponding pyplot functions as well, even if they are not used

Ideally this could be automated to catch both `pyplot -> axes` and `axes -> pyplot`?

File: examples/statistics/confidence_ellipse.py (hunk 1 above)
Comment (ianhi) [reply examples/statistics/confidence_ellipse.py:228]: good point. Although it seems that the backlinking does not work for classes like `Ellipse` or `Affine2D`. See https://51381-1385122-gh.circle-artifacts.com/0/doc/build/html/api/transformations.html#matplotlib.transforms.Affine2D where this example is not linked to :(

**Comment by ianhi** [comment]
> Generally, we add the corresponding pyplot functions as well

Done

File: examples/statistics/confidence_ellipse.py (hunk 1 above)
Comment (ianhi) [reply examples/statistics/confidence_ellipse.py:228]: I added that the Ellipse in the latest push, though the issue with it rendering will remain I suppose

**Comment by timhoffm** [comment]
> > Generally, we add the corresponding pyplot functions as well, even if they are not used

> Ideally this could be automated to catch both `pyplot -> axes` and `axes -> pyplot`?

Unfortunately this is all very manual: We're tuning our content so that sphinx-gallery does "the right thing". But sphinx-gallery does not know anything about pyplot <-> axes relations. And we cannot easily patch this in.

Probably the most promising direction of automation would be to customize the generation of the pyplot ReST code, so that it references the respective Axes minigallery; i.e.

~~~
matplotlib.pyplot.boxplot
=========================

.. currentmodule:: matplotlib.pyplot

.. autofunction:: boxplot

.. minigallery:: matplotlib.axes.Axes.boxplot
   :add-heading:
~~~

instead of `.. minigallery:: matplotlib.pyplot.boxplot`.

That way:

- we could limit ourselves to only giving Axes methods as references. (I see this as a plus, because in general we recommend the OO approach over pyplot.)
- Pyplot functions would still have a helpful minigallery.

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
I think you can continue that discussion in #19232.

**Pull request merged**